src/api/uploads/FolderUpload.js

```diff
@@ -158,12 +158,14 @@ class FolderUpload {
         successCallback: Function,
     }): Promise<any> {
         await this.folder.upload(this.destinationFolderId, errorCallback, true);
-        // Simulate BoxItem
-        successCallback([
-            {
-                id: this.folder.folderId,
-            },
-        ]);
+        // If the folder upload failed then a folderID will not be set
+        if (this.folder.folderId) {
```

Comment:
Could change this to `this.folder.getFolderId()` for consistency, even though they're synonymous.

Reply:
Yeah change it to that + make it a constant on top for re-usability below

Reply:
would go with the function, as that might have some value checking logic

Reply:
Actually thinking more about this, I am not sure if we should be relying on folder id being present or not to detect an error. It feels like assumption logic, which means that somewhere else in the code you have to make sure that folder id is not set for this condition to work

Reply:
I would think about adding try/catch in this function too, and throw a new error. In that way it will be explicit and just scoped to error handling.

Reply:
Something like

```js
    /**
     * Upload a folder
     *
     * @public
     * @param {string} parentFolderId
     * @param {Function} errorCallback
     * @param {boolean} isRoot
     * @returns {Promise}
     */
    async upload(parentFolderId: string, errorCallback: Function, isRoot: boolean = false) {
        this.parentFolderId = parentFolderId;

		try {
        	await this.createAndUploadFolder(errorCallback, isRoot);
        } catch (error) {
        	return;
        }
        
        this.addFilesToUploadQueue(this.getFormattedFiles(), noop, true);
        await this.uploadChildFolders(errorCallback);
    }
    
    /**
     * Create folder and add it to the upload queue
     *
     * @private
     * @param {Function} errorCallback
     * @param {boolean} isRoot
     * @returns {Promise}
     */
    createAndUploadFolder = async (errorCallback: Function, isRoot: boolean) => {
        await this.buildCurrentFolderFromEntry();

        let errorObj = {};

        try {
            const data = await this.createFolder();
            this.folderId = data.id;
        } catch (error) {
            if (error.code === ERROR_CODE_ITEM_NAME_IN_USE) {
                this.folderId = error.context_info.conflicts[0].id;
            } else {
            	if (isRoot) {
            		errorCallback(error);
                    throw error; // non-recoverable error, abort
                } else {
                	errorObj = { status: STATUS_ERROR, error };
                    errorCallback({ code: ERROR_CODE_CHILD_FOLDER_FAILED_UPLOAD });
                }
            }
        }
        
        // The root folder has already been added to the upload queue in ContentUploader
        if (isRoot) {
            return;
        }

        const folderObject: UploadItem = {
            extension: '',
            name: this.name,
            status: STATUS_COMPLETE,
            isFolder: true,
            size: 1,
            progress: 100,
            ...errorObj
        };

        this.addFolderToUploadQueue([folderObject]);
    };
```